### ISSUE TITLE
Block Toolbar: No confirmation for expanded / collapsed for screen readers

### ISSUE THREAD

**Issue by donalirl** [WordPress/gutenberg#22015]
**Describe the bug**
When working with blocks on a mobile web browser, additional block options are expanded by tapping on the ... in the top right corner. However, there is no label for expanded or collapsed, but this is necessary for visually-impaired users to know if the settings are expanded.

**To reproduce**
1. Edit any page or post with existing blocks
2. Using a screen reader, navigate to a block to edit it
3. The block toolbar has more options behind a .... Tap this
4. Observe that the screen reader does not recognize that the more options have been expanded

**Expected behavior**
I expected that the screen reader will say "expanded" when more options are expanded, and "collapsed" when it is closed. This action needs a label.

**Screenshots**
See this screenshot: https://d.pr/i/1SB1SO 

**Editor version (please complete the following information):**
WordPress version: 5.4
Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? Default

**Smartphone (please complete the following information):**
 - Device: iPod Touch
 - Browser: Safari Latest
 - Screen reader: Apple Voiceover

**Comment by afercia** [comment]
I'm not sure I fully understand this issue.

The "More options" button in the block toolbar uses an `aria-expanded` attribute to communicate its state. It also uses an `aria-haspopup` attribute to indicate it opens a menu that requires user activation. 

When the menu opens, focus is moved to the first item in the menu. This is the expected interaction for an [ARIA menu button](https://www.w3.org/TR/wai-aria-practices-1.1/#menubutton). Since focus is moved, the screen reader doesn't announce the `aria-expanded` value change: instead, it announces the new focused item, ant this is the expected behavior.

**Comment by afercia** [comment]
Unfortunately I don't own an Apple mobile device so I can't test it. I guess gestures may not honour constraining focus (actually: _tabbing_) within a specific UI. However, this isn't the point reported on this issue :)

**Comment by talldan** [comment]
I've left a comment in the #accessibility slack channel to see if we can get some further feedback on this one:
https://wordpress.slack.com/archives/C02RP4X03/p1613453140296500

Removed the `Needs Testing` label as `Needs Accessibility Feedback` seems like it covers everything.

**Comment by alexstine** [comment]
I have given this a try on my Mac and it is working fine. Focus remains in the menu and aria-expanded receives true/false based on state of "More" menu in the block toolbar.

I think it's good to close.
